File: leetcode/PalindromeNumber/isPalindrome.java
public class Solution {
    public boolean isPalindrome(int x) {
        // Start typing your Java solution below
        // DO NOT write main() function
        if (x < 0) return false;
        int cnt = 1;
        while (x / cnt >= 10) cnt *= 10;
        while (x > 0) {
            if (x / cnt != x % 10) return false;
            x %= cnt;
            x /= 10;
            cnt /= 100;
        }
        return true;
    }
}

File: leetcode/PalindromeNumber/isPalindrome.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        int y = 0;
        while (x > y) {
            y = y * 10 + x % 10;
            if (!y) return false;
            x /= 10;
        }
        return x == y || x == y / 10;
    }
};
